xchange-bitfinex/src/main/java/org/knowm/xchange/bitfinex/v2/BitfinexHmacSignature.java

```diff
@@ -0,0 +1,58 @@
+package org.knowm.xchange.bitfinex.v2;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import javax.ws.rs.HeaderParam;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.service.BaseParamsDigest;
+import org.knowm.xchange.utils.DigestUtils;
+import si.mazi.rescu.RestInvocation;
+
+public class BitfinexHmacSignature extends BaseParamsDigest {
+
+  private static final String UTF_8 = "UTF-8";
+  /**
+   * Constructor
+   *
+   * @param secretKeyBase64
+   * @throws IllegalArgumentException if key is invalid (cannot be base-64-decoded or the decoded
+   *     key is invalid).
+   */
+  private BitfinexHmacSignature(String secretKeyBase64) {
+
+    super(secretKeyBase64, HMAC_SHA_384);
+  }
+
+  public static BitfinexHmacSignature createInstance(String secretKeyBase64) {
+
+    return secretKeyBase64 == null ? null : new BitfinexHmacSignature(secretKeyBase64);
+  }
+
+  @Override
+  public String digestParams(RestInvocation i) {
+
+    // /api/${apiPath}${nonce}${JSON.stringify(body)}
+
+    String path = i.getPath();
+    try {
+      path = URLDecoder.decode(path, "utf8");
+    } catch (UnsupportedEncodingException ignored) {
+    }
```

Comment:
silent errors are evil, even the "its almost impossible" ones ;)
just surround the entire methods body with try - catch
you have a good catch that handles it in the end

Reply:
i do not see a reason to change this, sorry
basically, we try to url decode the "path", if it fails, so what, we simply use the "path" like it is
the worst thing in case of wrong "path" value would be, that we end up having an invalid signature, which would lead to a failed request on the server with an according error. but this scenario does not happen in all my tests. so what can be improved here ?

Reply:
ok, then at least log something in this catch